### PR TITLE
Add retry for uploading artifact

### PR DESCRIPTION
**Pull request by OpportunityLiu**
Related #12 

+ target updated to es2019 since it is fully supported by nodejs12
+ retry 3 times for each artifact with a failure of 5xx
+ add some debug messages, only shown by setting the secret `ACTIONS_STEP_DEBUG` to `true`.

### CODE REVIEW

**Comment by ncipollo** [comment]
Thanks for the pull request! Taking a look now.

File: __tests__/ArtifactUploader.test.ts [hunk 1]
```diff
@@ -134,4 +196,15 @@ describe('ArtifactUploader', () => {
     function mockListWithoutAssets() {
         listArtifactsMock.mockResolvedValue({ data: [] })
     }
+
+    function mockUploadArtifact(status: number = 200, failures: number = 0) {
+        uploadMock.mockImplementation(() => new Promise((resolve, reject) => {
+            if (failures > 0) {
+                failures--
```
Comment (ncipollo) [inline __tests__/ArtifactUploader.test.ts]: I think we can remove this line, it doesn't look like `failures` is used again within this scope.

File: tsconfig.json [hunk 2]
```diff
@@ -2,7 +2,7 @@
   "compilerOptions": {
     /* Basic Options */
     // "incremental": true,                   /* Enable incremental compilation */
-    "target": "es6",                          /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017', 'ES2018', 'ES2019' or 'ESNEXT'. */
+    "target": "es2019",                          /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017', 'ES2018', 'ES2019' or 'ESNEXT'. */
```
[the remark this answers is not in the document]
Comment (OpportunityLiu) [reply tsconfig.json]: No,the only improvment is used here is native async/await support.

I'll keep this, you can revert by yourself after merge.

File: __tests__/ArtifactUploader.test.ts (hunk 1 above)
Comment (OpportunityLiu) [reply __tests__/ArtifactUploader.test.ts]: It is just used by L202. The variable is caputured by reference.

I'll rewrite this part to make it more readable.

**Comment by ncipollo** [comment]
Oof, sorry when I squashed and merged it removed you as the author of the commit 😞 . If you would like credit I can add a contributors section to the read me.

**Pull request merged**